### ISSUE TITLE
[FR] Horizontal fractions

### ISSUE THREAD

**Issue by boromyr** [Qalculate/qalculate-gtk#510]
It would be a great step if in Qalculate fractions could be expressed in horizontal layout instead of the outdated single-line text input using `/` to divide. This, in addition to a better visualization of the input, would provide a significant speed of writing, eliminating the need to use parentheses on parentheses, and also a lower risk of errors due to incorrect or unnecessary parentheses.
This is a feature present both in my physical Casio and in Android apps.
For example, this could be how to display fractions in the future:
![image](https://github.com/Qalculate/qalculate-gtk/assets/52782032/7c1d3d66-712e-486d-8512-0526162bb627)
I am sure that this could be a novelty appreciated by any type of user.

**Comment by hanna-kn** [comment]
I agree that it would make the expression clearer, but I think that in many cases makes input less efficient (for simple numerators and denominators) and more complicated (it requires use of non-alphanumerical keys or pointing device the move out of numerator and denominator).

It would nevertheless be nice to provide it as an option. The problem is that it's quite complicated to implement.

**Comment by boromyr** [comment]
I don't think it complicates the input more than it simplifies it, you could just use tab, arrow keys or any other key, even customized, to navigate within the fraction, for example in [Microsoft Math](https://math.microsoft.com/en) you can create your own fraction:
![image](https://github.com/Qalculate/qalculate-gtk/assets/52782032/6d829730-3e1b-48c0-8ad4-dd143fae13e9)
and complete it in:
![image](https://github.com/Qalculate/qalculate-gtk/assets/52782032/fd320c87-daf9-414c-9b29-e1d7cc776557)
I don't know how complicated it might be to implement, but Qalculate is able to display the results in mathematical format.
![image](https://github.com/Qalculate/qalculate-gtk/assets/52782032/4537ab07-02e3-45cf-bece-5dac95e7770d)

**Comment by tethragon** [comment]
This feature request would be much appreciated by almost anyone. Many popular scientific calculators (physical calculators, web calculators and smartphone calculators) have this capability of "natural type". For a single developer I can see that this would be challenging/complicated. However, qalculate has far more and superior functions compared to most calculators out there. I wish there could be a way to implement this "natural type" to qalculate...
